### ISSUE TITLE
Overhaul of the alert and notices system

### ISSUE THREAD

**Issue by pehbeh** [osTicket/roadmap#64]
### Discussed in https://github.com/osTicket/roadmap/discussions/63

<div type='discussions-op-text'>

<sup>Originally posted by **pehbeh** April  8, 2023</sup>
I would like to suggest that the currently available options to be informed about new tickets or new messages from customers be revised and expanded. My idea here is that the settings can be **choose by each agent**. Currently, the settings from the department take effect. It would be great if the agent could decide this themselves. I would like to see the following notifications:

**Notify for**
- A new ticket
- A ticket is assigned to me			
- A ticket is assigned to someone else			
- I'm @mentioned in a ticket
- A ticket from my department has new updates (internal notes, due-date-updates...)
			
**Notify me when a customer replies**	 
- To an unassigned ticket			
- To a ticket assigned to me	
- To a ticket assigned by someone else		</div>

**Feature Summary**

Settings for notifications that can be set by each agent themselves

**How will it work once completed?**

Currently, the settings from the department take effect for every agent. It would be great if the agent could decide this themselves.

**Comment by RikkiMasters** [comment]
Is 2.0 still in development? I haven't seen any movement in quite some time.

**Comment by JediKev** [comment]
@RikkiMasters 

Of course. The code is not public yet so you won’t see much traction until we have an RC and make the code public.

Cheers.
